monkey/agent_plugins/exploiters/snmp/src/community_string_generator.py

```diff
@@ -0,0 +1,59 @@
+from itertools import chain
+from typing import Callable, Iterable, List
+
+from common.credentials import Identity, Password, Secret, Username
+from infection_monkey.exploit.tools import identity_type_filter, secret_type_filter
+from infection_monkey.propagation_credentials_repository import IPropagationCredentialsRepository
+
+
+class CommunityStringGenerator:
```

Comment:
I wonder if this really needs to be a class, or be coupled to `IPropagationCredentialsRepository` at all. If `generate_community_strings()` doesn't get called more than once, there's no need for it to maintain state. And if all we use from `PropagationCredentialsRepository` is the output of `get_credentials()`, then we could just call `get_credentials()` and pass its output into the utility function.